framework/TApplication.php

```diff
@@ -698,6 +699,60 @@ public function getModules()
 		return $this->_modules;
 	}
 
+	/**
+	 * Returns a list of application modules of a specific class.
+	 * Modules that have not been loaded yet are loaded.
+	 * @param string $type class name of the modules to look for.
+	 * @param bool $strict should just the class be found or subclasses of the class
+	 * @param bool $first return the first module found, default false.
+	 * @return array|object list of loaded application module of a specific class
+	 * or the first object of a specific class
+	 * @since 4.2.0
+	 */
+	public function getModulesByType($type, $strict = false, $first = false)
+	{
+		$m = [];
+		foreach ($this->_modules as $id => $module) {
+			if ($module === null) {
+				[$moduleClass, $initProperties, $configElement] = $this->_lazyModules[$id];
+				
+				$match = ($moduleClass === $type);
+				if (!$strict && !$match) {
+					$reflect = new ReflectionClass($moduleClass);
+					if ($reflect = $reflect->getParentClass()) {
+						do {
+							if ($reflect->getName() === $type) {
+								$match = true;
+								break;
+							}
+						} while ($reflect = $reflect->getParentClass());
+					}
+				}
+				
+				if ($match) {
+					$module = $this->internalLoadModule($id, true);
+					$module[0]->init($module[1]);
+					if ($first) {
+						return $module[0];
+					} else {
+						$m[] = $module[0];
+					}
+				}
+			} elseif (get_class($module) === $type || (!$strict && ($module->isa($type)))) {
+				if ($first) {
+					return $module;
+				} else {
+					$m[] = $module;
+				}
+			}
+		}
+		if ($first) {
```

Comment:
I's remove the $first parameter, making this method always return an array (possibly empty) of module ids.